File: android-core/src/main/java/com/mparticle/internal/KitContext.java
package com.mparticle.internal;


import android.app.Application;
import android.content.Context;
import android.content.ContextWrapper;


public class KitContext extends ContextWrapper {
    ApplicationContextWrapper applicationContextWrapper;

    public KitContext(Context base) {
        super(base);
        applicationContextWrapper = new ApplicationContextWrapper((Application) base.getApplicationContext());
    }

    @Override
    public Context getApplicationContext() {
        return applicationContextWrapper;
    }
}

File: android-core/src/main/java/com/mparticle/internal/KitContext.kt
package com.mparticle.internal

import android.app.Application
import android.content.Context
import android.content.ContextWrapper

open class KitContext(base: Context) : ContextWrapper(base) {
    private val applicationContextWrapper: ApplicationContextWrapper =
        ApplicationContextWrapper(base.applicationContext as Application)

    override fun getApplicationContext(): Context = applicationContextWrapper
}
